pkg/syncer/syncer.go

```diff
@@ -197,13 +207,28 @@ func StartSyncer(ctx context.Context, cfg *SyncerConfig, numSyncerThreads int, i
 		advancedSchedulingEnabled = true
 	}
 
+	dnsIP := ""
+	// Get the DNS IP. The DNS service cannot be recreated as existing dnsConfig won't be updated.
+	err = wait.PollImmediate(2*time.Second, time.Minute, func() (bool, error) {
+		ips, err := net.LookupIP(cfg.DNSServer)
+		if len(ips) == 0 || err != nil {
+			return false, nil //nolint:nilerr
```

Comment:
If err is not nil, I would store it in a variable declared outside the loop. Then, if the loop fails, include its details in the error returned on L221